tests/Paysera/WalletApi/MapperTest.php

```diff
@@ -3,41 +3,720 @@
     }
 
+    /**
+    * @dataProvider accessTokenDataProvider
+    */
+    public function testDecodeAccessToken($data, $expectedException, $expectedResult)
+    {
+        if ($expectedException !== null) {
+            $this->setExpectedException($expectedException);
+        }
+
+        $result = $this->mapper->decodeAccessToken($data);
+
+        if ($expectedResult !== null) {
+            $this->assertInstanceOf(Paysera_WalletApi_Entity_MacAccessToken::class, $result);
+            $this->assertEquals($expectedResult->getMacId(), $result->getMacId());
+            $this->assertEquals($expectedResult->getMacKey(), $result->getMacKey());
+            $this->assertEquals($expectedResult->getRefreshToken(), $result->getRefreshToken());
+        }
+    }
+
+    public function accessTokenDataProvider()
+    {
+        return [
+            'valid data' => [
+                'data' => [
+                    'token_type' => 'mac',
+                    'mac_algorithm' => 'hmac-sha-256',
+                    'expires_in' => 3600,
+                    'access_token' => 'test_token',
+                    'mac_key' => 'test_key',
+                    'refresh_token' => 'test_refresh_token',
+                ],
+                'expectedException' => null,
+                'expectedResult' => Paysera_WalletApi_Entity_MacAccessToken::create()
+                    ->setMacId('test_token')
+                    ->setMacKey('test_key')
+                    ->setRefreshToken('test_refresh_token')
+            ],
+            'invalid token type' => [
+                'data' => [
+                    'token_type' => 'invalid',
+                    'mac_algorithm' => 'hmac-sha-256',
+                    'expires_in' => 3600,
+                    'access_token' => 'test_token',
+                    'mac_key' => 'test_key',
+                    'refresh_token' => 'test_refresh_token',
+                ],
+                'expectedException' => InvalidArgumentException::class,
+                'expectedResult' => null
+            ],
+            'invalid mac_algorithm' => [
+                'data' => [
+                    'token_type' => 'mac',
+                    'mac_algorithm' => 'invalid',
+                    'expires_in' => 3600,
+                    'access_token' => 'test_token',
+                    'mac_key' => 'test_key',
+                    'refresh_token' => 'test_refresh_token',
+                ],
+                'expectedException' => InvalidArgumentException::class,
+                'expectedResult' => null
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider paymentProvider
+     */
+    public function testEncodePayment($payment, $expectedOutput, $expectedException = null)
+    {
+        if ($expectedException !== null) {
+            $this->setExpectedException($expectedException);
+        }
+        $result = $this->mapper->encodePayment($payment);
+        if ($expectedOutput !== null) {
```

Comment:
Perhaps the condition $expectedOutput === null is needed

Reply:
No. It's correct if expected result is not null then check response